platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java

```diff
@@ -183,6 +183,8 @@ public int getVersion() {
      */
     @Override
     public void serialize(final SerializableDataOutputStream out) throws IOException {
+        assert digestType != null;
```

Comment:
They are used elsewhere in data team code, and in the past there has been pain caused as a result. Use of assert requires us to catch `Throwable` instead of `Exception`, and has led to bugs where we don't expect `Throwable` and don't catch things the way we should.

I'd also advocate for something like `Objects.requeriesNonNull`.